src/replication.c

```diff
@@ -2658,9 +2691,32 @@ replicate(ISTGT_LU_DISK *spec, ISTGT_LU_CMD_Ptr cmd, uint64_t offset, uint64_t n
 
 	MTX_UNLOCK(&spec->rq_mtx);
 
-	extra_wait.tv_sec = extra_wait.tv_nsec = 0;
+	clock_gettime(CLOCK_MONOTONIC, &queued_time);
+
 	// now wait for command to complete
 	while (1) {
+		timesdiff(CLOCK_MONOTONIC, queued_time, now, diff);
+		count = 0;
+		for (i = 0; i < rcomm_cmd->copies_sent; i++) {
+			if (rcomm_cmd->resp_list[i].status &
+			    (RECEIVED_OK|RECEIVED_ERR|REPLICATE_TIMED_OUT))
+				count++;
+			else if (diff.tv_sec >= (time_t)io_max_wait_time) {
+				ASSERT(rcomm_cmd->resp_list[i].replica);
+				rcomm_cmd->resp_list[i].status |=
+				    REPLICATE_TIMED_OUT;
+
+				MTX_LOCK(&rcomm_cmd->resp_list[i].replica->r_mtx);
```

Comment:
When the replica gets disconnected due to network issue, there is a chance for replica being deleted after verifying status for 'RECEIVED_ERR' or 'RECEIVED_OK'.
So, lets take spec lock, and verify that replica that we want to disconnect is still part of the list, and do inform_mgmt_conn accordingly. There may not be a need for replica's r_mtx here.

Reply:
add log when the replica need to be disconnected due to timeout, and also, when the replica is not part of the list.

Reply:
ok